src/Alert.js

```diff
@@ -0,0 +1,96 @@
+import React, { PropTypes } from 'react';
+import classNames from 'classnames';
+import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
+
+const propTypes = {
+  className: PropTypes.any,
+  color: PropTypes.string,
+  onDismiss: PropTypes.func,
+  tag: PropTypes.oneOfType([PropTypes.func, PropTypes.string])
+};
+
+const defaultProps = {
+  color: 'warning',
+  tag: 'div'
+};
+
+export const Alert = (props) => {
+  const {
+    className,
+    tag: Tag,
+    color,
+    onDismiss,
+    ...attributes
+  } = props;
+
+  const classes = classNames(
+    className,
+    'alert',
+    `alert-${color}`,
+    { 'alert-dismissible': onDismiss }
+  );
+
+  return (
+    <Tag {...attributes} className={classes} role="alert">
+      { onDismiss ?
+        <button type="button" className="close" aria-label="Close" onClick={onDismiss}>
+          <span aria-hidden="true">&times;</span>
+        </button>
+        : null }
+      {props.children}
+    </Tag>
+  );
+};
+
+Alert.propTypes = propTypes;
+Alert.defaultProps = defaultProps;
+
+const wrapperPropTypes = {
+  dismissible: PropTypes.bool
+}
+
+const wrapperDefaultPropTypes = {
+  dismissible: false
+}
+
+class AlertWrapper extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      visible: true
+    }
+  }
+
+  onDismiss = () => {
+    this.setState({ visible: false });
```

Comment:
It could be a controlled and uncontrolled component. But I think that leads to inconsistency (updating a prop vs updating via setState) and extra docs. Preferring props over internal state is a general rule for this library. I built [component-template](https://github.com/reactstrap/component-template) with this sort of thing in mind. Someone could build `reactstrap-controlled-components`, which wrap these components and adds a controlled version of `Alert`. It's not the most convenient thing to do but it definitely enables more prescriptive ways of doing things. Even though that prescription is simplicity.